pkg/files/files.go

```diff
@@ -0,0 +1,40 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package files
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+func FilesInDir(path string) (map[string]string, error) {
+	files := make(map[string]string)
+	dir, err := os.ReadDir(path)
```

Comment:
On the same theme as my other comments, an empty `ConfigMap` (which is the default) is going to result in an empty map being returned here. This in turn will mean attempts will be made to create log URLs.

I think in this case, if there are file in the `ConfigMap`, then nil should be returned here as well.

Is there a test using the default `ConfigMap` that verifies that no log URLs are generated?  I want to make sure we don't event *attempt* to generate them if the configuration isn't there and correct.

Reply:
My bad, I missed that case initially.

I'd prefer returning an empty map instead of `nil` because a nil map cannot be modified. I think that it makes sense to add the logic to prevent log urls from being generated should be put in `cmd/sink/logs/logs.go` since this is functionality specific to the sink. The code in this file will be shared with the Api Server since it will also need to mount the `ConfigMap`. I can add a test to make sure that we don't attempt to add log urls to the database if the ConfigMap is empty

Reply:
> I'd prefer returning an empty map instead of nil because a nil map cannot be modified.

Why would you ever want to modify this map?

And the check to see if the URLs should be generated should be:
```
map != nil && map[LOG_URL] != nil
```

Reply:
> Why would you ever want to modify this map?

I don't know and that is the point! This function can be called in other places and we don't know what they will do with it

> And the check to see if the URLs should be generated should be:

There is already a check in `UrlBuilder.Urls` to not generate urls if the map is empty

https://github.com/kubearchive/kubearchive/blob/69a39e802f7d1d9d03054ba103cadc6807963a55/cmd/sink/logs/logs.go#L74


I've added an additional check to see if `map[LOG_URL]` exists

Reply:
>  I think that it makes sense to add the logic to prevent log urls from being generated should be put in cmd/sink/logs/logs.go since this is functionality specific to the sink.

There is already code in `main.go` to handle this, and this is why I am trying to get `sink.logUrlBuilder` to be nil in all these cases.
```
	// only write logs for k8s resources likes pods that have them and UrlBuilder is configured
	if sink.logUrlBuilder != nil && strings.ToLower(obj.GetKind()) == "pod" {
		sink.writeLogs(ctx, obj)
	}
```

Reply:
> Why would you ever want to modify this map?

I don't know where this will be called in the future an why since this is now it's own package. Additionally, I don't think it is worth adding extra logic to handle this when `len(map) == 0` catches both nil and empty map

> And the check to see if the URLs should be generated should be:

map != nil && map[LOG_URL] != nil


the check for log url cannot look like that. First of all, the zero value for strings in Go is `""` not `nil`. Secondly `map`'s type is not `map[string]string` it is `map[string]` interface so a type conversion is required. I've updated the logic and added a test to ensure that log urls are not generated if there is no `LOG_URL` field.

Reply:
I addressed the issues we discussed earlier. PTAL again